lib/src/component_declaration/redux_component.dart

```diff
@@ -0,0 +1,163 @@
+// Copyright 2016 Workiva Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+library over_react.component_declaration.redux_component;
+
+import 'dart:async';
+
+import 'package:built_redux/built_redux.dart';
+import 'package:meta/meta.dart';
+import 'package:over_react/over_react.dart';
+
+/// Builds on top of [UiProps], adding typed props for [Store] in order to integrate with `built_redux`.
+///
+/// Use with the over_react transformer via the `@Props()` ([annotations.Props]) annotation.
+abstract class ReduxUiProps<StoreT> extends UiProps {
+  String get _storePropKey => '${propKeyNamespace}store';
+
+  /// The prop defined by [StoresT].
+  ///
+  /// This object should either be an instance of [Store].
+  ///
+  /// __Instead of storing state within this component via `setState`, it is recommended that data be
+  /// pulled directly from these stores.__ This ensures that the data being used is always up to date
+  /// and leaves the state management logic to the stores.
+  StoreT get store => props[_storePropKey];
+  set store(StoreT value) => props[_storePropKey] = value;
+}
+
+/// Builds on top of [UiComponent], adding `built_redux` integration.
+///
+/// * Redux components are responsible for rendering application views and turning
+///   user interactions and events into [Action]s.
+/// * Redux components can use data from one or many [Store] instances to define
+///   the resulting component.
+///
+/// Use with the over_react transformer via the `@Component()` ([annotations.Component]) annotation.
+///
+/// > Related: [ReduxUiStatefulComponent]
+abstract class ReduxUiComponent<T extends ReduxUiProps> extends UiComponent<T> with _ReduxComponentMixin<T> {
+  @mustCallSuper
+  @override
+  void componentWillMount() {
+    super.componentWillMount();
+    _setUpSubs();
+  }
+
+  @mustCallSuper
+  @override
+  void componentWillReceiveProps(Map nextProps) {
+    super.componentWillReceiveProps(nextProps);
+    // Unbind store subs so they can be re-bound in componentDidUpdate
+    // once the new props are available, ensuring the values returned by [redrawOn]
+    // are not outdated.
+    _tearDownSubs();
```

Comment:
If the store prop is the same, we don't need new subs, right?

Reply:
It's not worth checking that, it doesn't take much to teardown and re-bind the subs

Reply:
I'm curious why it isn't worth it. It seems like you'd want to not force something that is not necessary.

Reply:
Tearing down and rebinding subscriptions is cheap, I don't think it's necessary to expend effort to avoid it.

Also, we we'd have to compare the return value of `redrawOn` and not just `props.store`, and we don't have access to that updated return value until `componentDidUpdate` (since that function can rely on values within `props`). So, we'd have to store the old values in here and then check them and null them out in componentDidUpdate.

In this case, unbinding/rebinding is more straightforward, and again, should cheap enough to be worth it.

Reply:
I'm confused about this, tracking _areStoreSubsBound and having all these teardown and setups it seems overly complicated to me. 
The reference to your store should never change, it seems to me that you would only setup subs on componentWillMount and remove them on componentWillUnmount.